src/libraries/System.Private.CoreLib/src/System/IO/Path.Unix.cs

```diff
@@ -9,9 +9,36 @@ namespace System.IO
 {
     public static partial class Path
     {
-        public static char[] GetInvalidFileNameChars() => new char[] { '\0', '/' };
+        public static char[] GetInvalidFileNameChars() => new char[] {'\0', '/'};
 
-        public static char[] GetInvalidPathChars() => new char[] { '\0' };
+        public static char[] GetInvalidPathChars() => new char[] {'\0'};
+
+        // Checks if the given path is available for use.
+        public static bool Exists([NotNullWhen(true)] string? path)
+        {
+            try
+            {
+                if (string.IsNullOrEmpty(path))
+                {
+                    return false;
+                }
+
+                ReadOnlySpan<char> fullPath = GetFullPath(path).AsSpan();
+                return Interop.Sys.LStat(fullPath, out Interop.Sys.FileStatus _) == 0;
+            }
+
+            catch (ArgumentException)
+            {
+            }
+            catch (IOException)
+            {
+            }
+            catch (UnauthorizedAccessException)
+            {
+            }
```

Comment:
similarly to the Windows part, the  `lstat` method does not throw but returns an error code instead.